dusty/systems/virtualbox/__init__.py

```diff
@@ -152,12 +166,30 @@ def initialize_docker_vm():
 
 @memoized
 def get_docker_vm_ip():
+    ssh_port = None
+    for line in _get_vm_config():
+        if line.startswith('Forwarding'):
```

Comment:
It's a line in the massive dump that VBox gives you for the machine config. We're taking advantage of the fact that Machine (or maybe just VBox) rigs up a localhost forwarding rule for SSH to the target machine. We use this to figure out WTF the target machine's actual IP is, which we need if we want to talk to it on any port other than the one that gets forwarded for us.